addons/account_peppol/models/res_partner.py

```diff
@@ -121,18 +121,18 @@ def _check_peppol_participant_exists(self, participant_info, edi_identification,
             # technically have an existing SMP url but they are not real Peppol participants
             return False
 
-        if check_company:
+        if get_provider:
             # if we are only checking company's existence on the network, we don't care about what documents they can receive
             if not service_href:
                 return True
 
-            access_point_contact = True
+            provider_name = True
             with contextlib.suppress(requests.exceptions.RequestException, etree.XMLSyntaxError):
                 response = requests.get(service_href, timeout=TIMEOUT)
                 if response.status_code == 200:
                     access_point_info = etree.fromstring(response.content)
-                    access_point_contact = access_point_info.findtext('.//{*}TechnicalContactUrl') or access_point_info.findtext('.//{*}TechnicalInformationUrl')
-            return access_point_contact
+                    provider_name = access_point_info.findtext('.//{*}ServiceDescription')
+            return provider_name
```

Comment:
That part is a bit unclear. `provider_name` being a boolean is weird... but then, it becomes a string... Seeing the name of the method, I have the feeling is should always return a boolean :thinking: Is it the case or not?
+ add a docstring

Reply:
It probably was at the beginning, then I imagine they wanted to get the some info while doing this call, so they hacked it, then the hack was there but never used, then I hacked it again :innocent: 
I'll try to clean that up.